src/game/Tactical/OppList.cc

```diff
@@ -3315,45 +3311,37 @@ UINT8 MovementNoise(SOLDIERTYPE* pSoldier) // XXX TODO000B
 	}
 
 	// if sneaker is moving through water
-	if ( bInWater )
+	// DeepWater() must be checked first because Water() matches all types of water
+	if (DeepWater(pSoldier->sGridNo))
 	{
-		iStealthSkill -= 10; // 10% penalty
+		iStealthSkill -= 20;
 	}
-	else if (DeepWater( pSoldier->sGridNo ) )
+	else if (Water(pSoldier->sGridNo))
 	{
-		iStealthSkill -= 20; // 20% penalty
+		iStealthSkill -= 10;
 	}
 
-	if ( pSoldier->bDrugEffect[ DRUG_TYPE_ADRENALINE ] )
-	{
-		// minus 3 percent per bonus AP from adrenaline
-		iStealthSkill -= 3 * pSoldier->bDrugEffect[ DRUG_TYPE_ADRENALINE ];
-	}
+	// minus 3 percent per bonus AP from adrenaline
+	iStealthSkill -= 3 * pSoldier->bDrugEffect[ DRUG_TYPE_ADRENALINE ];
 
-	/*
-	// if sneaker is too eager and impatient to "do it right"
-	if ((pSoldier->bTrait == OVER_ENTHUS) || (pSoldier->bAttitude == AGGRESSIVE))
-	{
-		ubStealthSkill -= 10;	// 10% penalty
-	}*/
-
-	iStealthSkill = __max( iStealthSkill, 0 );
+	iStealthSkill = std::max(iStealthSkill, 0);
 
+	UINT8 ubVolume;
 	if (!pSoldier->bStealthMode)	// REGULAR movement
 	{
-		ubMaxVolume = MAX_MOVEMENT_NOISE - (iStealthSkill / 16); // 9 - (0 to 6) => 3 to 9
+		UINT8 ubMaxVolume = MAX_MOVEMENT_NOISE - (iStealthSkill / 16); // 9 - (0 to 9) => 0 to 9
 
-		if (bInWater)
+		if (Water(pSoldier->sGridNo))
 		{
 			ubMaxVolume++; // in water, can be even louder
 		}
 		switch (pSoldier->usAnimState)
 		{
 			case CRAWLING:
-				ubMaxVolume -= 2;
+				ubMaxVolume = ubMaxVolume <= 2 ? 0 : ubMaxVolume - 2;
```

Comment:
Would it make sense to use `INT32` for `ubVolume` and `ubMaxVolume` as well and just use [`std::clamp`](https://en.cppreference.com/w/cpp/algorithm/clamp) in the end to limit the result to a `UINT8` value again?

Reply:
A simple `std::max(0, ubMaxVolume - 2)` would work as well.